.github/workflows/check_smoke_tests.yml

```diff
@@ -0,0 +1,96 @@
+name: check smoke tests
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  smoke_tests:
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Check out repository
+      uses: actions/checkout@v4
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+
+    - name: Set up Docker
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y docker-compose
+
+    - name: Start services with docker-compose
+      run: docker-compose up -d
+    
+    - name: Wait for RL to become healthy
+      run: |
+        # Wait for the RL Service to be healthy
+        TRIES=20
+        COUNT=0
+        until curl -s http://localhost:8080/ | grep "OK"; do
+          echo "Waiting for RL Service to become healthy... Attempt $((COUNT+1)) of $TRIES"
+          sleep 5
+          COUNT=$((COUNT+1))
+          if [ "$COUNT" -ge "$TRIES" ]; then
+            echo "RL Service did not become healthy in time."
+            exit 1
+          fi
+        done
+
+    - name: Smoke tests
+      run: |
+        # Run smoke tests and print the response
+        JSON_BODY_1='{"record": {"birth_date": "2053-11-07", "sex": "M", "mrn": "1234567890", "name":[{"family":"Shepard", "given":["John"]}]}}'
+        JSON_BODY_2='{"algorithm": "dibbs-enhanced", "record": {"birth_date": "2000-12-06", "sex": "M", "mrn": "9876543210", "name":[{"family":"Smith", "given":["William"]}]}}'
+
+        #basic tests
+        RESPONSE_1=$(curl -s -X POST http://localhost:8080/link \
+        -d "$JSON_BODY_1" \
+        -H "Content-Type: application/json")
+
+        echo "Response: $RESPONSE_1"
+        echo "$RESPONSE_1" | grep -q '"is_match":\s*false' 
```

Comment:
Cool use of grep.  I believe `jq` is also available by default on ubuntu-latest if you wanted to use that.